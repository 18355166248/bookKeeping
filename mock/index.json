{
  "constList": [
    {
      "id": 1,
      "title": "买了airPods",
      "price": 1098,
      "date": "2019-09-17",
      "type": "outcome",
      "cid": 1
    },
    {
      "id": 2,
      "title": "买了苹果11",
      "price": 7999,
      "date": "2019-09-27",
      "type": "outcome",
      "cid": 1
    },
    {
      "id": 3,
      "title": "去日月光打车",
      "price": 50,
      "date": "2019-08-27",
      "type": "outcome",
      "cid": 2
    },
    {
      "id": 4,
      "title": "8月份工资",
      "price": 11111,
      "date": "2019-09-15",
      "type": "income",
      "cid": 3
    }
  ],

  "categoryList": [
    {
      "id": 1,
      "name": "电子",
      "type": "income",
      "iconName": "money-collect"
    },
    {
      "id": 2,
      "name": "出行",
      "type": "income",
      "iconName": "car"
    },
    {
      "id": 3,
      "name": "旅游",
      "type": "income",
      "iconName": "sketch"
    },
    {
      "id": 4,
      "name": "工资",
      "type": "outcome",
      "iconName": "dollar"
    },
    {
      "id": 5,
      "name": "购物",
      "type": "income",
      "iconName": "shopping"
    },
    {
      "id": 6,
      "name": "生活",
      "type": "income",
      "iconName": "home"
    },
    {
      "id": 7,
      "name": "其他",
      "type": "income",
      "iconName": "cloud"
    },
    {
      "id": 8,
      "name": "兼职",
      "type": "outcome",
      "iconName": "credit-card"
    },
    {
      "id": 9,
      "name": "理财",
      "type": "outcome",
      "iconName": "property-safety"
    }
  ]
}
