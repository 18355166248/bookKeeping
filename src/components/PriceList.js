import React, { Component } from 'react'
import PriceItem from './priceItem/priceItem'
import PropTypes from 'prop-types'

class PriceList extends Component {
  render() {
    return (
      <div style={{ borderTop: '1px solid #e8e8e9' }}>
        {this.props.list.length > 0 ? (
          this.props.list.map((v, i) => <PriceItem {...v} key={i} />)
        ) : (
          <div
            className="ac m_t_10 m_t_10 empty_data"
            style={{ fontSize: '18px' }}
          >
            暂无数据
          </div>
        )}
      </div>
    )
  }
}

PriceList.propTypes = {
  list: PropTypes.array.isRequired
}
export default PriceList
