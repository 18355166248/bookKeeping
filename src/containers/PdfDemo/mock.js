export const data = {
  doctor: { en_US: 'Alex Jiang', zh_CN: '蒋益' },
  createFileTime: '2020-03-25',
  patientAllergyRecordList: [
    {
      allergyDegreeName: '轻度',
      allergyReactionName: '胸闷',
      allergySourceName: '大豆蛋白',
      allergyType: 9
    },
    {
      allergyDegreeName: '轻度',
      allergyReactionName: '面色苍白',
      allergySourceName: '(乙10%)兰索拉唑肠溶胶囊 (达克普隆) 30mg',
      allergyType: 1
    }
  ],
  patient: {
    age: '30Y2M2D',
    birthday: { en_US: '1990/01/23', zh_CN: '1990/01/23' },
    gender: { en_US: 'Male', zh_CN: '男' },
    mrn: '0020000983',
    name: { en_US: 'CHEN/LEI', zh_CN: '陈磊' }
  },
  diagnosisTreatmentPlanList: [{ diagnosisTreatmentPlanName: '38牙齿拔牙' }],
  deptInfo: {
    campus: { en_US: 'JIH', zh_CN: 'JIH' },
    medicalInstitution: {
      en_US: 'Jiahui International Hospital (Shanghai)',
      zh_CN: '上海嘉会国际医院'
    },
    specialty: { en_US: 'Dentistry (JIH)', zh_CN: '嘉会口腔科' }
  },
  medicalHistory: {
    dentalPastHistory: '口腔病史',
    diagnosisMedicalRecordCheckNormalList: [
      {
        checkNormalResult: '检查结果1',
        checkNormalSymptoms: '检查症状1',
        diagnosisPosition: '38'
      },
      {
        checkNormalResult: '检查结果2',
        checkNormalSymptoms: '检查症状2',
        diagnosisPosition: ''
      }
    ],
    diagnosisMedicalRecordCheckRayList: [
      { checkRaySymptoms: '辅助检查1', diagnosisPosition: '38' },
      { checkRaySymptoms: '辅助检查2', diagnosisPosition: '' }
    ],
    diagnosisMedicalRecordDiagnosisList: [
      { diagnosisDesc: '诊断1', diagnosisPosition: '38' },
      {
        diagnosisDesc: '诊断2',
        diagnosisPosition: '11,12,13,14,15,16,17,18,21,22,23,24,25,26,27,28'
      }
    ],
    diagnosisMedicalRecordDisposeList: [
      { diagnosisPosition: '38', dispose: '处置1' },
      { diagnosisPosition: '', dispose: '处置2' },
      { diagnosisPosition: '', dispose: '处置3' }
    ],
    doctorAdviceList: [
      {
        itemName: { en_US: 'Complicated extraction', zh_CN: '复杂拔牙/颗' },
        itemType: 4,
        total: '1'
      },
      {
        itemName: {
          en_US: 'CT maxillofacial w/o contrast (t)',
          zh_CN: '颌面部CT检查, 平扫 (T)'
        },
        itemType: 4,
        total: '1',
        unit: ''
      },
      {
        doseUnit: 'MG',
        everyDayDose: '200.00',
        frequencyName: { en_US: 'BID', zh_CN: '每日两次' },
        itemName: {
          en_US: 'Metronidazole tablets 200mg',
          zh_CN: '甲硝唑片 200mg'
        },
        itemType: 3,
        total: '1',
        unit: 'BOX',
        usageName: { en_US: 'PO', zh_CN: '口服' }
      }
    ],
    historySocialHabitsFlag: true,
    mainComplaint: '左侧下颌牙齿疼痛两天',
    pastIllnessHistoryBloodFlag: true,
    pastIllnessHistoryBloodList: [
      {
        amount: '1ml',
        bloodDate: '2017/1/1',
        bloodType: '04',
        bloodTypeName: '血浆Plasma',
        createTime: '2020-03-20 19:56:30',
        effect: '反应1',
        episodeId: '01OP20001782',
        hasEffect: '1',
        isValid: '1',
        keyCategory: 'blood',
        keyCategoryNameCN: '输血史',
        keyCategoryNameEN: 'blood',
        patientId: '0020000983',
        remark: '备注1',
        visitType: '2'
      },
      {
        amount: '4',
        bloodDate: '',
        bloodType: '01',
        bloodTypeName: '全血Whole Blood',
        createTime: '2020-03-20 19:56:47',
        effect: '',
        episodeId: '01OP20001782',
        hasEffect: '0',
        isValid: '1',
        keyCategory: 'blood',
        keyCategoryNameCN: '输血史',
        keyCategoryNameEN: 'blood',
        patientId: '0020000983',
        remark: '备注2',
        visitType: '2'
      }
    ],
    pastIllnessHistoryDiseaseFlag: true,
    pastIllnessHistoryDiseaseList: [
      {
        createTime: '2020-03-20 22:06:51',
        diseaseCode: '',
        diseaseEnd: '',
        diseaseName: '疾病',
        diseaseProcess: '',
        diseaseStart: '',
        episodeId: '01OP20001782',
        isInfect: '0',
        isValid: '1',
        keyCategory: 'disease',
        keyCategoryNameCN: '疾病史',
        keyCategoryNameEN: 'disease',
        patientId: '0020000983',
        remark: '',
        treatment: '',
        visitType: '2'
      },
      {
        createTime: '2020-03-20 22:07:46',
        diseaseCode: '',
        diseaseEnd: 'Now',
        diseaseName: '高血压',
        diseaseProcess: '',
        diseaseStart: '2017/1/1',
        episodeId: '01OP20001782',
        isInfect: '0',
        isValid: '1',
        keyCategory: 'disease',
        keyCategoryNameCN: '疾病史',
        keyCategoryNameEN: 'disease',
        patientId: '0020000983',
        remark: '备注',
        treatment: '治疗措施',
        visitType: '2'
      },
      {
        createTime: '2020-03-20 22:07:46',
        diseaseCode: '',
        diseaseEnd: 'Now',
        diseaseName: '糖尿病',
        diseaseProcess: '',
        diseaseStart: 'Unknown',
        episodeId: '01OP20001782',
        isInfect: '0',
        isValid: '1',
        keyCategory: 'disease',
        keyCategoryNameCN: '疾病史',
        keyCategoryNameEN: 'disease',
        patientId: '0020000983',
        remark: '',
        treatment: '疗',
        visitType: '2'
      }
    ],
    pastIllnessHistoryDrinkFlag: true,
    pastIllnessHistoryDrinkList: [
      {
        amount: '20',
        createTime: '2020-03-20 21:58:21',
        eTime: '',
        episodeId: '01OP20001782',
        isDrink: '2',
        keyCategory: 'drink',
        keyCategoryNameCN: '饮酒史',
        keyCategoryNameEN: 'drink',
        month: '0',
        patientId: '0020000983',
        remark: '备注2',
        sTime: '2017/4/7',
        visitType: '2',
        year: '0'
      }
    ],
    pastIllnessHistoryFamilyFlag: true,
    pastIllnessHistoryFamilyList: [
      {
        createTime: '2020-03-20 19:58:23',
        diseaseName: '疾病1',
        episodeId: '01OP20001782',
        isCancer: '0',
        isValid: '1',
        keyCategory: 'family',
        keyCategoryNameCN: '家庭史',
        keyCategoryNameEN: 'family',
        onsetAge: '21',
        patientId: '0020000983',
        relationCode: '2',
        relationNameChn: '父亲Father',
        relationNameEng: '',
        remark: '备注1',
        visitType: '2'
      },
      {
        createTime: '2020-03-20 19:58:23',
        diseaseName: '疾病2',
        episodeId: '01OP20001782',
        isCancer: '0',
        isValid: '1',
        keyCategory: 'family',
        keyCategoryNameCN: '家庭史',
        keyCategoryNameEN: 'family',
        onsetAge: '23',
        patientId: '0020000983',
        relationCode: '3',
        relationNameChn: '母亲Mother',
        relationNameEng: '',
        remark: '备注2',
        visitType: '2'
      }
    ],
    pastIllnessHistoryOperationFlag: true,
    pastIllnessHistoryOperationList: [
      {
        createTime: '2020-03-20 15:40:51',
        episodeId: '01OP20001782',
        isValid: '1',
        keyCategory: 'operation',
        keyCategoryNameCN: '手术史',
        keyCategoryNameEN: 'operation',
        operationDate: '2017/1/1',
        operationExplain: '',
        operationName: '阑尾切除',
        patientId: '0020000983',
        remark: '',
        visitType: '2'
      },
      {
        createTime: '2020-03-20 19:56:12',
        episodeId: '01OP20001782',
        isValid: '1',
        keyCategory: 'operation',
        keyCategoryNameCN: '手术史',
        keyCategoryNameEN: 'operation',
        operationDate: '',
        operationExplain: '',
        operationName: '手术1',
        patientId: '0020000983',
        remark: '备注',
        visitType: '2'
      }
    ],
    pastIllnessHistoryOtherFlag: true,
    pastIllnessHistoryOtherList: [
      {
        createTime: '2020-03-20 19:57:52',
        episodeId: '01OP20001782',
        isValid: '1',
        keyCategory: 'other',
        keyCategoryNameCN: '其他病史',
        keyCategoryNameEN: 'other',
        patientId: '0020000983',
        remark: 'QITA 1',
        visitType: '2'
      },
      {
        createTime: '2020-03-20 19:57:52',
        episodeId: '01OP20001782',
        isValid: '1',
        keyCategory: 'other',
        keyCategoryNameCN: '其他病史',
        keyCategoryNameEN: 'other',
        patientId: '0020000983',
        remark: 'QITA 2',
        visitType: '2'
      }
    ],
    pastIllnessHistorySmokeFlag: true,
    pastIllnessHistorySmokeList: [
      {
        amount: '20',
        createTime: '2020-03-20 21:58:21',
        eTime: 'Now',
        episodeId: '01OP20001782',
        isSmoke: '1',
        keyCategory: 'smoke',
        keyCategoryNameCN: '吸烟史',
        keyCategoryNameEN: 'smoke',
        month: '0',
        patientId: '0020000983',
        remark: '备注1',
        sTime: '2017/1/1',
        unit: '支Cigarettes',
        visitType: '2',
        year: '0'
      }
    ],
    pastIllnessHistorySportFlag: true,
    pastIllnessHistorySportList: [
      {
        createTime: '2020-03-20 19:57:52',
        episodeId: '01OP20001782',
        keyCategory: 'sport',
        keyCategoryNameCN: '运动史',
        keyCategoryNameEN: 'sport',
        patientId: '0020000983',
        remark: '备注3',
        times: '2',
        visitType: '2'
      }
    ],
    pastIllnessHistoryTravelFlag: true,
    pastIllnessHistoryTravelList: [
      {
        createTime: '2020-03-20 19:57:52',
        episodeId: '01OP20001782',
        isValid: '1',
        keyCategory: 'travel',
        keyCategoryNameCN: '旅行史',
        keyCategoryNameEN: 'travel',
        patientId: '0020000983',
        remark: '意大利',
        visitType: '2'
      },
      {
        createTime: '2020-03-20 19:57:52',
        episodeId: '01OP20001782',
        isValid: '1',
        keyCategory: 'travel',
        keyCategoryNameCN: '旅行史',
        keyCategoryNameEN: 'travel',
        patientId: '0020000983',
        remark: '挪威',
        visitType: '2'
      }
    ],
    pastIllnessHistoryWorkFlag: true,
    pastIllnessHistoryWorkList: [
      {
        createTime: '2020-03-20 19:57:52',
        episodeId: '01OP20001782',
        isValid: '1',
        keyCategory: 'work',
        keyCategoryNameCN: '职业',
        keyCategoryNameEN: 'work',
        month: '0',
        patientId: '0020000983',
        remark: '',
        visitType: '2',
        work: 'IT1',
        year: '1'
      },
      {
        createTime: '2020-03-20 19:57:52',
        episodeId: '01OP20001782',
        isValid: '1',
        keyCategory: 'work',
        keyCategoryNameCN: '职业',
        keyCategoryNameEN: 'work',
        month: '1',
        patientId: '0020000983',
        remark: '',
        visitType: '2',
        work: 'IT2',
        year: '0'
      }
    ],
    presentIllnessHistory:
      '3周前左侧下牙常有阵发性疼痛，未治疗，两天前开始抽痛，放散到整个左下颌及耳部，不能咀嚼，遂来就诊。',
    vitalSigns:
      '体温 Body temp:38.0℃ 血压 BP:80.0/120.0 mmHg 脉搏 Pulse:89.0 bpm 呼吸 BR:21 min 身高 Height:178.0 cm 体重 Weight:89.0 KG '
  },
  registerDate: '2020-03-20 15:38:43'
};
