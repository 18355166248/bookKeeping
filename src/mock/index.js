export const constList = [
  {
    id: 1,
    title: '买了airPods',
    price: 1098,
    date: '2019-09-17',
    category: {
      id: 1,
      name: '电子',
      type: 'outcome',
      iconName: 'money-collect'
    }
  },
  {
    id: 2,
    title: '买了苹果11',
    price: 7999,
    date: '2019-09-27',
    category: {
      id: 1,
      name: '电子',
      type: 'outcome',
      iconName: 'money-collect'
    }
  },
  {
    id: 3,
    title: '去日月光打车',
    price: 50,
    date: '2019-08-27',
    category: {
      id: 1,
      name: '出行',
      type: 'outcome',
      iconName: 'car'
    }
  },
  {
    id: 4,
    title: '8月份工资',
    price: 13578,
    date: '2019-09-15',
    category: {
      id: 1,
      name: '收入',
      type: 'income',
      iconName: 'wallet'
    }
  }
]
