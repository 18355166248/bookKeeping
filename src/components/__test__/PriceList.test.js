import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PriceList from '../PriceList'
import PriceItem from '../priceItem/priceItem'
import { constList, categoryList } from '../../mock'

const list = constList.map(v => {
  v.category = categoryList.find(v1 => v1.id === v.cid)
  return v
})
let wrapper
describe('测试 PriceList 组件', () => {
  beforeEach(() => {
    wrapper = shallow(<PriceList list={list} />)
  })
  it('匹配快照', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('PriceItem组件应该渲染了4个', () => {
    expect(wrapper.find(PriceItem).length).toBe(4)
  })
  it('PriceList组件空数据的时候是否显示暂无数据', () => {
    wrapper.setProps({list: []})
    expect(wrapper.find('.empty_data').exists()).toBeTruthy()
  })
})
