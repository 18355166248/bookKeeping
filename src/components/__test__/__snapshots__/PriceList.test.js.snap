// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 PriceList 组件 匹配快照 1`] = `
<div
  style={
    Object {
      "borderTop": "1px solid #e8e8e9",
    }
  }
>
  <withRouter(PriceItem)
    category={
      Object {
        "iconName": "money-collect",
        "id": 1,
        "name": "电子",
        "type": "outcome",
      }
    }
    cid={1}
    date="2019-09-17"
    id={1}
    index={0}
    key="0"
    price={1098}
    title="买了airPods"
  />
  <withRouter(PriceItem)
    category={
      Object {
        "iconName": "money-collect",
        "id": 1,
        "name": "电子",
        "type": "outcome",
      }
    }
    cid={1}
    date="2019-09-27"
    id={2}
    index={1}
    key="1"
    price={7999}
    title="买了苹果11"
  />
  <withRouter(PriceItem)
    category={
      Object {
        "iconName": "car",
        "id": 2,
        "name": "出行",
        "type": "outcome",
      }
    }
    cid={2}
    date="2019-08-27"
    id={3}
    index={2}
    key="2"
    price={50}
    title="去日月光打车"
  />
  <withRouter(PriceItem)
    category={
      Object {
        "iconName": "wallet",
        "id": 3,
        "name": "收入",
        "type": "income",
      }
    }
    cid={3}
    date="2019-09-15"
    id={4}
    index={3}
    key="3"
    price={11111}
    title="8月份工资"
  />
</div>
`;
